src/client/kinto_http.rs

```diff
@@ -108,28 +116,35 @@ pub fn get_changeset(
         server, bid, cid, expected, since_param
     );
     info!("Fetch {}...", url);
-    let response = Request::get(Url::parse(&url)?).send()?;
+    let response = requester
+        .get(Url::parse(&url)?)
+        .map_err(|_err| KintoError::HTTPBackendError())?;
 
     if !response.is_success() {
         // Try to parse the server error response into JSON.
         // See https://docs.kinto-storage.org/en/stable/api/1.x/errors.html#error-responses
-        let info: ErrorResponse = match response.json() {
+        let info: ErrorResponse = match serde_json::from_slice(&response.body) {
             Ok(v) => v,
-            Err(_) => return Err(KintoError::UnexpectedResponse { response }),
+            Err(_) => return Err(KintoError::UnexpectedResponse { url, response }),
         };
 
         // Error due to the client. The request must be modified.
-        if response.is_client_error() {
-            return Err(KintoError::ClientRequestError { response, info });
+        if (400..500).contains(&response.status) {
+            return Err(KintoError::ClientRequestError {
+                url,
+                response,
+                info,
+            });
         }
 
-        if response.is_server_error() {
+        if (500..600).contains(&response.status) {
```

Comment:
Since we have `is_sucess()` maybe it wouldn't harm to have `is_client_error()` and `is_server_error()`